frontend/app/components/auth/auth.tsx

```diff
@@ -16,19 +17,22 @@ import { OAuth } from './components/oauth';
 import { messages } from './auth.messsages';
 import { useDropdown } from './auth.hooks';
 import { getProviders, getTokenInvalidReason } from './auth.utils';
-import { emailSignin, verifyEmailSignin, anonymousSignin } from './auth.api';
+import { emailSignin, verifyEmailSignin, anonymousSignin, verifyTelegramSignin } from './auth.api';
+import { StoreState } from 'store';
 
 import styles from './auth.module.css';
 
 export function Auth() {
   const intl = useIntl();
   const dispatch = useDispatch();
+  const telegramParams = useSelector((s: StoreState) => s.telegramParams);
   const [oauthProviders, formProviders] = getProviders();
 
   // UI State
   const [isLoading, setLoading] = useState(false);
   const [view, setView] = useState<typeof formProviders[number] | 'token'>(formProviders[0]);
-  const [ref, isDropdownShown, toggleDropdownState] = useDropdown(view === 'token');
+  const [isTelegramShown, toggleTelegram] = useState(false);
```

Comment:
This state seems unnecessary. We can `view` as an indicator of the current view. Just add 'telegram' option to it.

Reply:
Fixed in 589659d